doc/platform/common_config.md

```diff
@@ -39,30 +42,59 @@ The functional requirements include :
+|-- x86_64-broadcom_b88 -- broadcom-sonic-td4.config.bcm
+|-- x86_64-broadcom_b96 -- broadcom-sonic-th.config.bcm
+|-- x86_64-broadcom_b97 -- broadcom-sonic-th2.config.bcm
+|-- x86_64-broadcom_b98 -- broadcom-sonic-th3.config.bcm
+|-- x86_64-broadcom_b99 -- broadcom-sonic-th4.config.bcm
+|-- x86_64-broadcom_f90 -- broadcom-sonic-th5.config.bcm
 ```
 ## 3 Design Detail
 The main change of the design is in the SYNCD docker syncd/scripts/syncd_init_common.sh script along with common config being created in the device/broadcom/x86_64-broadcom_common/ folder. The design standardize the common file name as broadcom-sonic-{$chip_id}.config.bcm . Also, in the SYNCD docker-syncd-brcm.mk, we extern the common config directory path /usr/share/sonic/device/x86_64-broadcom_common from host to docker for script reference.
 
 The design change for syncd_init_common.sh is targeted on the config_syncd_bcm() which is only used for BRCM switch chip. In the config_syncd_bcm() function, it will load the common silicon config.bcm file by looking for a *.bcm file in the  /usr/share/sonic/device/x86_64-broadcom_common/x86_64-broadcom_{$chip_id} folder where the $chip_id is the upper 3 nibbles of the switching silicon's device ID as listed in SDK/include/soc/devids.h. Then, merge common broadcom-sonic-{$chip_name}.config.bcm file with the existing platform specific config.bcm file in which the duplicate configuration entries in the platform specific file will override entries in the common broadcom-sonic-{$chip_name}.config.bcm file. 
 
-Since the platform specific config.bcm is read only in docker, the design copies the platform specific config.bcm and sai.profile to /tmp for handling common config merge process. The /tmp/sai.profile will be modified to point to the merged config.bcm under/tmp directory. The following switch initialization will reference to the new merged config.bcm pointed by updated sai.profile from the /tmp directory.
+In the updated 0.2 version, the script will check the platform specific config file. If it is yml based, it will merge common broadcom-sonic-{$chip_name}.config.bcm file with the existing platform specific config.yml file. 
 
+In order to support common config overwriting the platform config feature in updated version 0.2. We propose a option for supporting properties of common config overwriting to the Platform config. The properties of common config in the overwrite section have the high priority than the platform specfic config and will merge and overwrite to the platform config. As the following example, the properties of common config in the overwrite section will overwirte the platform specifc config during the merged process. Note, the Overwrite Section should be located after normal section in the common config file, otherwise the logic will overwrite all properties
+
+```
+/device/broadcom/x86_64-broadcom_common/x86_64-broadcom_b27/broadcom-sonic-td3.config.bcm
+[Normal Section]
```

Comment:
I think the name of the sections can be made more intuitive, since this section is meant to have higher precedence over the the platform counterpart, can we call it as the [Low Inheritance Precedence] section? Whereas the other section is meant to be of lower precedence where the platform file attributes will overwrite, then call the other section as [High Inheritance Precedence]

Reply:
You mean to have [overwrite section] Key changed to [High Inheritance Precedence] ? I am ok with the Key.

Reply:
Correct, overwrite section -> High Inheritance Precedence, normal section -> Low Inheritance Precedence

Reply:
got it